app/models/g5_authenticatable/user.rb

```diff
@@ -47,6 +47,23 @@ def update_roles_from_auth(auth_data)
       end
     end
 
+    def clients
+      return G5Updatable::Client.all if has_global_role
+      G5Updatable::Client.where(id: client_roles.map(&:resource_id))
+    end
+
+    def client_roles
+      G5Authenticatable::Role.joins('INNER JOIN g5_updatable_clients ON g5_updatable_clients.id = g5_authenticatable_roles.resource_id')
+        .where('g5_authenticatable_roles.resource_type = ?', G5Updatable::Client.name)
+    end
+
+    def has_global_role
+      has_role?(:admin) ||
```

Comment:
Jonathan, I'm not sure but I think you should be able to do this using a ``Policy`` object, 
Look at ``/app/policies/g5_authenticatable/base_policy.rb``